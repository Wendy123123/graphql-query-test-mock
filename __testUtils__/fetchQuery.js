// @flow
type Operation = {
  name: string,
  text: string
};

export const GRAPHQL_API_URL = 'http://localhost/graphql';

export type FetchQueryConfig = {
  query: string,
  name: string,
  variables?: { [key: string]: mixed }
};

export function fetchQuery(
  operation: Operation,
  variables?: { [key: string]: mixed }
): Promise<*> {
  const config: FetchQueryConfig = {
    query: operation.text,
    name: operation.name,
    variables
  };

  return fetch(GRAPHQL_API_URL, {
    method: 'POST',
    headers: {
      'content-type': 'application/json'
    },
    body: JSON.stringify(config)
  }).then(response => response.json());
}
