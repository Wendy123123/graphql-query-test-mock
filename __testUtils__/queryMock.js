// @flow
import { QueryMock } from '../src/index';

export const queryMock = new QueryMock();
