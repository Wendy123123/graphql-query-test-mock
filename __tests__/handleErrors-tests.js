// @flow
import { fetchQuery } from '../__testUtils__/fetchQuery';
import { queryMock } from '../__testUtils__/queryMock';


describe('handleErrors', () => {
  describe('printNoMockFoundError', () => {
    it('should throw an error when there is no query name', async () => {
      try {
        const res = await fetchQuery({
          text: 'query { id }',
        });
      } catch (error) {
        expect(error.message).toMatchSnapshot();
        return;
      };
      throw new Error('No error was raised');
    });

    it('should throw an error when there is no mocked query', async () => {
      try {
        const res = await fetchQuery({
          text: 'query TestQuery { id }',
        });
      } catch (error) {
        expect(error.message).toMatchSnapshot();
        return;
      };
      throw new Error('No error was raised');
    });

    it('should throw an error when there is 1 matching query name', async () => {
      queryMock.mockQuery({
        name: 'TestQuery',
        data: {},
        variables: {
          variable1: 'VARIABLE_1',
        },
      });

      try {
        const res = await fetchQuery({
          text: 'query TestQuery { id }',
        });
      } catch (error) {
        expect(error.message).toMatchSnapshot();
        return;
      };
      throw new Error('No error was raised');
    });

    it('should throw an error when there are multiple matching query names', async () => {
      queryMock.mockQuery({
        name: 'TestQuery',
        data: {},
        variables: {
          variable1: 'VARIABLE_1',
        },
      });
      queryMock.mockQuery({
        name: 'TestQuery',
        data: {
          test: 'DATA 2',
        },
        variables: {
          variable2: 'VARIABLE_2',
        },
      });

      try {
        const res = await fetchQuery({
          text: 'query TestQuery { id }'
        });
      } catch (error) {
        expect(error.message).toMatchSnapshot();
        return;
      };
      throw new Error('No error was raised');
    });

    it('should throw an error when there is 1 non-matching query names', async () => {
      queryMock.mockQuery({
        name: 'TestQuery1',
        data: {},
      });

      try {
        const res = await fetchQuery({
          text: 'query TestQuery2 { id }'
        });
      } catch (error) {
        expect(error.message).toMatchSnapshot();
        return;
      };
      throw new Error('No error was raised');
    });

    it('should throw an error when there are multiple non-matching query names', async () => {
      queryMock.mockQuery({
        name: 'TestQuery1',
        data: {},
      });

      queryMock.mockQuery({
        name: 'TestQuery2',
        data: {},
      });

      queryMock.mockQuery({
        name: '', // unknown
        data: {},
      });

      try {
        const res = await fetchQuery({
          text: 'query TestQuery3 { id }'
        });
      } catch (error) {
        expect(error.message).toMatchSnapshot();
        return;
      };
      throw new Error('No error was raised');
    });

    it('should throw an error when there are both matching and non-matching query names', async () => {
      queryMock.mockQuery({
        name: 'TestQuery1',
        data: {},
        variables: {
          variable1: 'VARIABLE_1',
        },
      });
      queryMock.mockQuery({
        name: 'TestQuery2',
        data: {},
      });

      try {
        const res = await fetchQuery({
          text: 'query TestQuery1 { id }'
        });
      } catch (error) {
        expect(error.message).toMatchSnapshot();
        return;
      };
      throw new Error('No error was raised');
    });
  });
});
