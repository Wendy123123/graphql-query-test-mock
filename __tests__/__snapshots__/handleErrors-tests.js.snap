// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleErrors printNoMockFoundError message should show an error when there are both matching and non-matching query names 1`] = `
"Could not find matching mock for operation \\"TestQuery1\\" with variables: {}
Make sure you have mocked the query you are making.

=== Currently mocked \\"TestQuery1\\" queries ===
Query \\"variables\\": {\\"variable1\\":\\"VARIABLE_1\\"}
Diff of \\"variables\\":
[32m- Expected[39m
[31m+ Received[39m

[32m- Object {[39m
[32m-   \\"variable1\\": \\"VARIABLE_1\\",[39m
[32m- }[39m
[31m+ Object {}[39m

=== Other mocked queries ===
- \\"TestQuery2\\"

"
`;

exports[`handleErrors printNoMockFoundError message should show an error when there are multiple matching query names 1`] = `
"Could not find matching mock for operation \\"TestQuery\\" with variables: {}
Make sure you have mocked the query you are making.

=== Currently mocked \\"TestQuery\\" queries ===
Query \\"variables\\": {\\"variable1\\":\\"VARIABLE_1\\"}
Diff of \\"variables\\":
[32m- Expected[39m
[31m+ Received[39m

[32m- Object {[39m
[32m-   \\"variable1\\": \\"VARIABLE_1\\",[39m
[32m- }[39m
[31m+ Object {}[39m

Query \\"variables\\": {\\"variable2\\":\\"VARIABLE_2\\"}
Diff of \\"variables\\":
[32m- Expected[39m
[31m+ Received[39m

[32m- Object {[39m
[32m-   \\"variable2\\": \\"VARIABLE_2\\",[39m
[32m- }[39m
[31m+ Object {}[39m

"
`;

exports[`handleErrors printNoMockFoundError message should show an error when there are multiple non-matching query names 1`] = `
"Could not find matching mock for operation \\"TestQuery3\\" with variables: {}
Make sure you have mocked the query you are making.

=== All mocked queries ===
- \\"TestQuery1\\"
- \\"TestQuery2\\"
- \\"unknown\\"

"
`;

exports[`handleErrors printNoMockFoundError message should show an error when there is 1 matching query name 1`] = `
"Could not find matching mock for operation \\"TestQuery\\" with variables: {}
Make sure you have mocked the query you are making.

=== Currently mocked \\"TestQuery\\" queries ===
Query \\"variables\\": {\\"variable1\\":\\"VARIABLE_1\\"}
Diff of \\"variables\\":
[32m- Expected[39m
[31m+ Received[39m

[32m- Object {[39m
[32m-   \\"variable1\\": \\"VARIABLE_1\\",[39m
[32m- }[39m
[31m+ Object {}[39m

"
`;

exports[`handleErrors printNoMockFoundError message should show an error when there is 1 non-matching query names 1`] = `
"Could not find matching mock for operation \\"TestQuery2\\" with variables: {}
Make sure you have mocked the query you are making.

=== All mocked queries ===
- \\"TestQuery1\\"

"
`;

exports[`handleErrors printNoMockFoundError message should show an error when there is no mocked query 1`] = `
"Could not find matching mock for operation \\"TestQuery\\" with variables: {}
Make sure you have mocked the query you are making.

=== No query was mocked ===

"
`;

exports[`handleErrors printNoMockFoundError message should show an error when there is no query name 1`] = `"Could not find query name."`;
